### ISSUE TITLE
Bluetooth pairing failure

### ISSUE THREAD

**Issue by JitteryDoodle** [bretterer/home-assistant-rivian#158]
Hello,

I recently got a BT adapter which appears to work, except when I attempt to pair. I'm getting this error:
```
2024-05-29 03:49:08.856 homeassistant bluetoothd[3079]: src/device.c:new_auth() No agent available for request type 2
2024-05-29 03:49:08.857 homeassistant bluetoothd[3079]: device_confirm_passkey: Operation not permitted
```
This is on HAOS 12.3 (2024.5.5 Core), with integration version 1.2.0. Any ideas?

**Comment by natekspencer** [comment]
Have you added the Bluetooth adapter to Home Assistant via the Bluetooth integration? This looks like an error from core vs anything from this integration

**Comment by natekspencer** [comment]
Ok cool, thanks for the sanity check. Again, the previous errors are not from this integration specifically though. So can you enable debug logging on the Rivian integration and see if it is making it to that code and/or where it is failing specifically.

**Comment by natekspencer** [comment]
Also, what BT adapter are you using? It will need at least Bluetooth 4.2 otherwise the secure pairing won't complete and will likely fail with an "Authentication Failed" message.

**Comment by natekspencer** [comment]
Your signal strength is extremely weak `(RSSI: -90)` which is likely contributing to your issues.

From https://www.mokoblue.com/measures-of-bluetooth-rssi/:
• A more strong connection is between -30 to -55
• A strong connection starts from -55 to -67
• A terrible connection starts from -80 to -90
• An unusable connection starts from -90 and below

You'll need to find a way to get it closer or, alternatively, use an ESP32 with bluetooth proxy setup. https://github.com/bretterer/home-assistant-rivian/issues/157#issuecomment-2145838321

**Comment by billchurch** [comment]
The ESP32 bluetooth proxy was the trick for me... however I had to turn off bluetooth on my HA device itself before it started working... And honestly I've kept it off now and exclusively use the ESP32 bluetooth proxy. I just made a stick, joined it to my wifi and then plugged it into the car to get the pairing to work. After that it kind of hung (on the rivian side) while paring but then I noticed in HA my buttons for the Rivian features became available.
